src/dfx/src/commands/canister/install.rs

```diff
@@ -97,123 +97,125 @@ pub async fn exec(
 
     let pull_canisters_in_config = get_pull_canisters_in_config(env)?;
 
-    if let Some(canister) = opts.canister.as_deref() {
-        if pull_canisters_in_config.contains_key(canister) {
-            bail!(
-                "{0} is a pull dependency. Please deploy it using `dfx deps deploy {0}`",
-                canister
-            );
-        }
-        let config = env.get_config();
-        let is_remote = config
-            .as_ref()
-            .map_or(Ok(false), |config| {
-                config
-                    .get_config()
-                    .is_remote_canister(canister, &network.name)
-            })
-            .unwrap_or(false);
-        if is_remote {
-            bail!("Canister '{}' is a remote canister on network '{}', and cannot be installed from here.", canister, &network.name)
-        }
-
-        let canister_id =
-            Principal::from_text(canister).or_else(|_| canister_id_store.get(canister))?;
+    let config = env.get_config_or_anyhow()?;
+    let config_interface = config.get_config();
+    let env_file = config.get_output_env_file(opts.output_env_file)?;
 
+    if let Some(canister) = opts.canister.as_deref() {
         let arguments_from_file = opts
             .argument_file
             .map(|v| arguments_from_file(&v))
             .transpose()?;
         let arguments = opts.argument.as_deref();
-        let arguments = arguments_from_file.as_deref().or(arguments);
-
+        let argument_from_cli = arguments_from_file.as_deref().or(arguments);
         let arg_type = opts.argument_type.as_deref();
-        let canister_info = config.as_ref()
-            .ok_or_else(|| anyhow!("Cannot find dfx configuration file in the current working directory. Did you forget to create one?"))
-            .and_then(|config| CanisterInfo::load(config, canister, Some(canister_id)));
-        if let Some(wasm_path) = opts.wasm {
-            // streamlined version, we can ignore most of the environment
-            let mode = mode.context("The install mode cannot be auto when using --wasm")?;
-            let install_args = || blob_from_arguments(arguments, None, arg_type, &None);
-            install_canister(
-                env,
-                &mut canister_id_store,
-                canister_id,
-                canister_info.ok().as_ref(),
-                Some(&wasm_path),
-                install_args,
-                Some(mode),
-                call_sender,
-                opts.upgrade_unchanged,
-                None,
-                opts.yes,
-                None,
-                opts.no_asset_upgrade,
-            )
-            .await
-            .map_err(Into::into)
+
+        // opts.canister is canister_id
```

Comment:
If the intent of this comment is to say that `canister_id` in `Some(canister_id)` and `opts.canister` are the same thing, this would be the order in which to say it
```suggestion
        // canister_id is opts.canister
```